src/com/sony/qcrilam/BootReceiver.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2018-2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Rewritten in Kotlin by Pavel Dubrova <pashadubrova@gmail.com>
+ */
+
+package com.sony.qcrilam
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+
+class BootReceiver : BroadcastReceiver() {
+    private var TAG = "QcRilAm-BootReceiver"
+
+    override fun onReceive(context: Context, intent: Intent) {
```

Comment:
Oh, this function has an overridable variant with non-nullable arguments? Or has it always been non-null, with covariance allowing a nullable-argument function to override it?

I vaguely remember _some function_ (not necessarily `BroadcastReceiver.onReceive`) having a "special" case where `intent` is `null`. Can't find it anymore though.

Reply:
@bartcubbins Ah, you removed it again.

Indeed, there's no special magic here. Since this is implementing a java class, all that's generated is extra code to do the null-validation for us, instead of printing `Log.w(TAG, "BOOT_COMPLETE NULL intent");`.

```java
    public void onReceive(Context context, Intent intent) {
        Intrinsics.checkParameterIsNotNull(context, "context");
        Intrinsics.checkParameterIsNotNull(intent, "intent");
        context.startService(new Intent(context, QcRilAmService.class));
    }
```

That will throw a nice exception for us instead. Which I'm a fan of, instead of silently muffling this away behind a _warning_ and not starting the service. Though we can easily get away with a null `intent`, only `context` is required.